x-pack/plugins/cases/server/services/cases/index.ts

```diff
@@ -564,11 +564,16 @@ export class CasesService {
     try {
       this.log.debug(`Attempting to POST a new case`);
       const transformedAttributes = transformAttributesToESModel(attributes);
+
+      transformedAttributes.attributes.total_alerts = -1;
```

Comment:
I think we should be consistent as to where we are defying the variables even if it is not perfect. `transformNewCase` seems more appropriate as is where we define all the other attributes. It seems that the `postNewCase`'s argument definition forces consumers (the case client for example) to think about the structure in ES and provide the data accordingly. In the future, it may be nice to refactor and move the logic of the `transformNewCase` inside the `postNewCase` function so consumers will only pass `CasePostRequest` objects.

Reply:
`transformNewCase` might make sense but in the future, because right now the return type is `CaseAttributes` which does not have `total_alerts` or `total_comments`. These fields are internal so that is why I thought it would make sense to have them closer to the transformation to ESModel.

> It seems that the postNewCase's argument definition forces consumers (the case client for example) to think about the structure in ES and provide the data accordingly.

But right now we don't want them to think about anything. The fields will quietly be there until there is an equivalent in `CaseAttributes`.

Reply:
I missed that the `transformNewCase` returns a `CaseAttributes` object which does not include `total_alerts` and `total_comments` as they are `ESCaseAttributes`. Make sense to me then. Also, because `transformAttributesToESModel` is also used for the update we do not want to put it there, correct? @jonathan-buttner What do you think?

Reply:
Yeah good points, leaving it here is fine.